text/0000-stabilize-arbitrary-self-types.md

```diff
@@ -32,11 +32,15 @@ When declaring a method, users can declare the type of the `self` receiver to be
 The `Receiver` trait is simple and only requires to specify the `Target` type to be resolved to:
 
 ```rust
-trait Receiver {
+/// # SAFETY
+/// - the type needs to be ABI-compatible with a fat pointer
+unsafe trait Receiver {
     type Target: ?Sized;
 }
 ```
 
+It is unsafe because implementers need to guarantee that the type is ABI-compatible with a fat pointer.
```

Comment:
Are you sure? - I think this is only necessary if `DispatchFromDyn` is implemented.

Even then, `DispatchFromDyn` doesn't need to be `unsafe` because the compiler checks _whether_ the type is ABI-compatible with a fat pointer, and refuses to do method dispatch on it otherwise.

This is a safe choice but currently results in poor diagnostics!

Reply:
If you agree: Let me know if you'd like me to merge this PR and then you'd like to adjust this later, that's fine with me!

Reply:
OK I'll merge this then fix it up later - I'm not at work tomorrow so after that.